### PR TITLE
chore: Add first unit tests

### PR DESCRIPTION
**Pull request by Gmerold**
- Adds some first unit tests for the CI
- Updates workflow file to run CI checks on every push

### CODE REVIEW

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -8,8 +8,6 @@ on:
     branches:
       - main
   push:
-    branches:
-      - main
```
Comment (ghislainbourgeois) [inline .github/workflows/main.yml:12]: Why is this removed? Won't this prevent this workflow from running on the main branch when PRs are merged?

Comment (gab-arrobo) [reply .github/workflows/main.yml:12]: @ghislainbourgeois, this change will make the workflow run for all branches (not just the `main` branch). I added those two lines last night to follow the same template we are using for all of the other repos. I am ok either way (keeping these two lines or removing them as done in this PR). What do you think? or what is your preference?

Comment (ghislainbourgeois) [reply .github/workflows/main.yml:12]: Oh right, I forgot the `push:` line above. That works for me then!

**Review by ghislainbourgeois — APPROVED** [review]

File: common/events_test.go [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+func TestString_InvalidEventTypeExitsWithLogMsg(t *testing.T) {
+	var INVALID_EVENT EventType = 0x0
+	var logBuf bytes.Buffer
+	assert := assert.New(t)
+	patchedExit := func(int) { panic("Dummy exit function") }
+
+	logger.AppLog.Logger.SetOutput(&logBuf)
+	logger.AppLog.Logger.ExitFunc = patchedExit
+
+	assert.Panics(func() { INVALID_EVENT.String() }) //nolint:govet
```
[the remark this answers is not in the document]
Comment (Gmerold) [reply common/events_test.go]: Done.

**Review by gab-arrobo — APPROVED** [review]
+1

**Review by thakurajayL — APPROVED** [review]
